ТЗ 16. Add-docker.

NurievLinar opened the pull request.

NurievLinar committed:
ТЗ 16. Add-docker.

NurievLinar committed:
ТЗ 16. Add-docker. Поправка properties.

NurievLinar committed:
ТЗ 16. Add-docker. Поправка properties.

NurievLinar committed:
ТЗ 16. Add-docker. Правка ошибок.

NurievLinar committed:
ТЗ 16. Add-docker. Исправление ошибок для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Исправление ошибок checkstyle для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Исправление ошибок checkstyle для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Исправление ошибок для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Правка кода для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Правка кода для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Правка кода для прохождения тестов GitHub.

NurievLinar committed:
ТЗ 16. Add-docker. Правка кода для прохождения тестов GitHub.

NurievLinar commented:
Сергей, доброе утро! Извините за мой ужасный код, но сегодня в полночь был жёсткий дедлайн. Не знаю смогут ли меня пропустить дальше или отчислят, академы я потратил - семейные обстоятельства.
Долго бился над тестами, локально всё проходит, в докере и на гитхабе тесты валятся.

sergeymamonov reviewed (changes requested):
Класс BookingServiceImpl проверка на 43 строке - ее лучше перенести в сервис gateway, так как именно он валидирует входные данные.

Проверку параметров пагинации и статуса (BookingServiceImpl строки 157 и 167) тоже лучше делать в  сервисе gateway.

PaginationException не место в основном сервисе.

Из проекта нужно удалить все TODO.

ItemServiceImpl и ItemRequestServiceImpl - убрать проверку валидации.

BadRequestException - не используется в основном сервисе.

NurievLinar committed:
ТЗ 16. Add-docker. Исправление замечаний ревью.
- проверки пагинации и статуса перенесены в gateway.
- удалены все TODO
- удалены или перемещены из server в gateway exception.

NurievLinar committed:
ТЗ 16. Add-docker. CheckStyle

NurievLinar committed:
ТЗ 16. Add-docker. CheckStyle Tests.

NurievLinar committed:
ТЗ 16. Add-docker. CheckStyle Tests.
